### ISSUE TITLE
Avoid crashing the node process when determining our external IP.

### ISSUE THREAD

**Issue by defunctzombie** [CodaProtocol/coda#3039]
My typical setup for nodes involves machines which set inbound and outbound firewall rules to drop all. I then whitelist ports and IP ranges as necessitated by the project and risk assessments. This includes outbound traffic as it can aid with host security.

I configured my firewall rules to allow port 8303 and 8302 (in and out) per the documentation. However, when I launched my node, it still crashed. This seems to have happened because it tried to talk to another node that was using ports other than the documented defaults.

It would be nice if my node did not crash when it tried and failed to talk to another node - instead simply moving on to connect to other peers.

```
coda daemon -peer hello-coda.o1test.net:8303


  ☠  Coda Daemon crashed. The Coda Protocol developers would like to know why!

  Please:
    Open an issue:
      https://github.com/CodaProtocol/coda/issues/new

    Briefly describe what you were doing, and include the last 20 lines from .coda-config/coda.log.
    And then paste the following:

    (monitor.ml.Error
  ("connection attempt timeout" 50.16.229.140:443 src/tcp.ml:120:11)
  ("Raised at file \"src/error.ml\" (inlined), line 9, characters 14-30"
    "Called from file \"src/error.ml\" (inlined), line 5, characters 2-50"
    "Called from file \"src/tcp.ml\", line 120, characters 11-76"
    "Called from file \"src/job_queue.ml\" (inlined), line 131, characters 2-5"
    "Called from file \"src/job_queue.ml\", line 170, characters 6-47"
    "Caught by monitor coda"))
```

**Comment by cmr** [comment]
This request isn't a peer connection, it's the request we make to `api.ipify.org` to determine our own external IP. This request can be circumvented by passing `-external-ip YOUR_IP` to the daemon. We definitely shouldn't crash in this case.

**Comment by cmr** [comment]
Docs should mention firewall configuration/this issue with HTTPS in particular

**Comment by o1pranay** [comment]
closing this out - resolved in #3072

**Comment by o1pranay** [comment]
thank you @defunctzombie for reporting this issue and for the thorough description :)
